prometheus/src/main/scala/fr/davit/akka/http/metrics/prometheus/PrometheusRegistry.scala

```diff
@@ -52,7 +50,7 @@ object PrometheusRegistry {
   * Prometheus registry
   * For metrics naming see [https://prometheus.io/docs/practices/naming/]
   */
-class PrometheusRegistry(settings: HttpMetricsSettings, val underlying: CollectorRegistry) extends HttpMetricsRegistry {
+class PrometheusRegistry(settings: HttpMetricsSettings, val underlying: CollectorRegistry, val prometheusSettings: PrometheusSettings = PrometheusSettings()) extends HttpMetricsRegistry {
```

Comment:
Currently the `HttpMetricsSettings` settings object is only used in the Prometheus implementation anyway, but I think this will complicate the usage of the settings objects once the settings are used by more than one implementation.

Unless the idea is to make `PrometheusSettings` a trait as well and have all users of the library provide a custom implementation of the settings trait combined with all the specific settings traits they need. This would also complicate setting defaults for the library, unless the default setting implementation includes _all_ settings traits.

This largely depends on what the plan with the other backend implementations is.